### PR TITLE
Feat/update e2e test

### PR DESCRIPTION
**Pull request by isaackps**
## Summary

This story is about updating existing e2e tests, both synpress and testcafe, from Goerli testnets to local.
While converting it to local, i encounter some issues and complications.
Below describes the findings and would like to have an opinion on what is the best course of action to take as a team. 

## Issues/ findings
Testcafe:
- we are using fixtures that are issued on goerli testnets, when goerli sunsets at the end of the year, all the fixtures will be unusable.
- If we change them to local, everytime when we run test, we will need to re-issue the document as the local env does not have the title-escrow, token registry, document store. We will need to re-deploy them into our environment as well.
- Test run for longer time and takes longer to debug UI in future.

Synpress:
- Currently we are forking goerli testnet and running it on local, this means that we have the token registry, title escrow factory and document store already deployed and we do not need to do so on local. This method works as long as the goerli testnet is available. But goerli has been deprecated and will be gone by the end of the year, so we are trying to localise it. 

## Recommendations
Testcafe:
- Option 1: We fork testnet and use it as our local as everytime we need to do the set up for all the files and run blockchain then do the test, which should not be the case as testcafe should be testing UI only and shouldn't need the whole setup.
- Option 2: We do not move it to local, we change the fixtures' testnets. Meaning, we need to re-issue the documents in the new testnet. By doing so, our test speed will remain the same and easier to debug UI in future. Also by doing so, if the testnet sunsets, we will need to do the same thing again with the new testnet. but by looking at the testnets lifecycle, this is a 1-time-thing every 3-4 years. 

Synpress:
- Will continue to localise it

### CODE REVIEW

**Comment by simboonlong** [comment]
Testcafe:
- that's true, tests on CI will take longer.

Synpress:
- ~~not sure if can skip the document wrap process by providing the wrapped local fixtures directly instead, then do the relevant deployment + issue. need to try out.~~ can skip
- not true on tests will fail. dns should not matter as we are skipping entire verification [here](https://github.com/TradeTrust/tradetrust-website/blob/65141e22eb4858af6d2f930def55b5784624221c/src/sagas/certificate.ts#L34-L36) for local.

**Comment by isaackps** [comment]
@simboonlong 
Synpress:
- i just run it with out the dns and wrapping and it is working. 
- yes u r right, it should work. should be previously i missed out something. 

will update the findings.

File: .gitignore [hunk 1]
```diff
@@ -15,5 +15,7 @@ yarn.lock
 /artifacts
 /tests/e2e/screenshots
 /tests/e2e/videos
+/tests/e2e/fixtures/wrapped
+/tests/e2e/fixtures/unwrapped
 .lighthouseci
```
Comment (simboonlong) [inline .gitignore]: let's not programmatically wrap? we can prepare the wrapped fixtures directly instead

File: tests/e2e/setup-contracts.mjs [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
-const ADDRESS_EXAMPLE_2 = "0xcDFAcbb428DD30ddf6d99875dcad04CbEFcd6E60";
+const ADDRESS_EXAMPLE_2 = "0xcdfacbb428dd30ddf6d99875dcad04cbefcd6e60";
+const ADDRESS_EXAMPLE_3 = "0x391aFf3942857a10958425FebF1fC1938D9F5AE7";
 
 const oaCLI_PATH = "open-attestation";
 
+export const contractAddress = {
+  TitleEscrowFactory: "0x9Eb613a88534E2939518f4ffBFE65F5969b491FF",
+  Deployer: "0x9eBC30E7506E6Ce36eAc5507FCF0121BaF7AeA57",
+  TokenImplementation: "0xE5C75026d5f636C89cc77583B6BCe7C99F512763",
```
Comment (simboonlong) [inline tests/e2e/setup-contracts.mjs]: what's the use of `Deployer` and `TokenImplementation` here?

File: tests/e2e/setup-contracts.mjs [hunk 3]
```diff
@@ -18,31 +27,42 @@ const merkleRootToMint = {
...
     {
       // Transfer Holder
-      merkleRoot: "0x1c92119b530b5028e8cb45e74624a5f52a1ae84d68e1c165776813a84722807a",
+      merkleRoot: "0x3a7436eac11bb9a98f75de7743e70e99b4f1a2625b372449b83979cf8e9b5183",
       ...defaultToken,
     },
   ],
 };
 
+shell.exec(`${oaCLI_PATH} deploy document-store "My Document Store" -n local -k ${ACCOUNT_KEY}`);
```
Comment (simboonlong) [inline tests/e2e/setup-contracts.mjs]: in synpress flow, we are only testing token registry / ebl fixtures, no need for deploy doc store

File: tests/e2e/specs/surrender-1-reject.spec.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 describe("Surrender-Reject", () => {
   context("Surrender", () => {
     it("should go to verify page, upload a file, connect to wallet and surrender a document successfully", () => {
+      cy.switchMetamaskAccount(1);
       cy.visit("/verify");
```
Comment (simboonlong) [inline tests/e2e/specs/surrender-1-reject.spec.js]: there's no need to edit any tests at all if im not wrong

**Comment by simboonlong** [comment]
i suggest split this into 2 prs? this 1 let's make synpress "local" without the fork to goerli bc.

once done, you might have a clearer idea on how to update testcafe tests against local fixtures.

File: .gitignore (hunk 1 above)
Comment (isaackps) [reply .gitignore]: erm, i'll update this, we do not need to wrapped the documents anymore.

File: tests/e2e/setup-contracts.mjs (hunk 2 above)
Comment (isaackps) [reply tests/e2e/setup-contracts.mjs]: erm, we dont need it already, will clean up.

File: tests/e2e/setup-contracts.mjs (hunk 3 above)
Comment (isaackps) [reply tests/e2e/setup-contracts.mjs]: yup, will clean up.

File: tests/e2e/specs/surrender-1-reject.spec.js (hunk 4 above)
Comment (isaackps) [reply tests/e2e/specs/surrender-1-reject.spec.js]: need to, as the account was not switch back in my testing.

**Comment by isaackps** [comment]
> i suggest split this into 2 prs? this 1 let's make synpress "local" without the fork to goerli bc.
> 
> once done, you might have a clearer idea on how to update testcafe tests against local fixtures.

alright, sure

File: .gitignore [hunk 5]
```diff
@@ -16,4 +16,4 @@ yarn.lock
 /tests/e2e/screenshots
 /tests/e2e/videos
 .lighthouseci
-lighthouseReport
\ No newline at end of file
+lighthouseReport
```
Comment (simboonlong) [inline .gitignore]: nitpick: omit this change? there's no change

File: tests/e2e/setup-contracts.mjs [hunk 6]
```diff
@@ -1,12 +1,14 @@
...
-const TOKEN_REGISTRY_ADDRESS = "0xf18CD26780B6D3589371fb0b3fE8E2a513D6Fdc6"; // Pre-deployed on goerli using token-registry or oa-cli
+const TOKEN_REGISTRY_ADDRESS = "0x9Eb613a88534E2939518f4ffBFE65F5969b491FF";
 const ADDRESS_EXAMPLE_1 = "0xe0a71284ef59483795053266cb796b65e48b5124";
-const ADDRESS_EXAMPLE_2 = "0xcDFAcbb428DD30ddf6d99875dcad04CbEFcd6E60";
+const ADDRESS_EXAMPLE_2 = "0xcdfacbb428dd30ddf6d99875dcad04cbefcd6e60";
 
 const oaCLI_PATH = "open-attestation";
 
+const TitleEscrowFactory = "0x63A223E025256790E88778a01f480eBA77731D04";
+
```
Comment (simboonlong) [inline tests/e2e/setup-contracts.mjs]: nitpick:
```suggestion
const TitleEscrowFactoryAddress = "0x63A223E025256790E88778a01f480eBA77731D04";

```

File: tests/e2e/specs/surrender-1-reject.spec.js (hunk 4 above)
Comment (simboonlong) [reply tests/e2e/specs/surrender-1-reject.spec.js]: as discussed, can leave code comments to why

File: .gitignore (hunk 5 above)
Comment (isaackps) [reply .gitignore]: erm but should have a newline at the end of file right?

File: tests/e2e/setup-contracts.mjs (hunk 6 above)
Comment (isaackps) [reply tests/e2e/setup-contracts.mjs]: ok sure

File: .gitignore (hunk 5 above)
Comment (simboonlong) [reply .gitignore]: should be handled by prettier, may want to check if it's default rules are been enforced or not

File: tests/e2e/specs/transfer-holder.spec.js [hunk 7]
```diff
@@ -1,5 +1,9 @@
 import { ACCOUNT_2 } from "../utils";
 
+before(() => {
+  cy.switchMetamaskAccount(1);
+});
```
Comment (simboonlong) [inline tests/e2e/specs/transfer-holder.spec.js]: comment?

**Review by simboonlong — APPROVED** [review]
can merge once the tests passed

File: .gitignore (hunk 5 above)
Comment (isaackps) [reply .gitignore]: yup it does enforce it, but it does not enforce in this file. so i manually add it in. is that not ok? if so i can remove it.

Comment (simboonlong) [reply .gitignore]: oh, then leave it lo. probably we should bake the check in pipeline then next time

**Pull request merged**